### PR TITLE
Setup 'empty' oathkeeper to act as a noop proxy towards workday

### PR DESCRIPTION
**Pull request by ZzAve**
With this setup, Workday is preparing for authentication and authorization outisde of the app itself, but managed through Ory's Oathkeeper, and identity aware access proxy.

```
internet --> oathkeeper --> workday
         <---           <---
```

To give some context to what is happening and why. We have a grant master plan to take all authentication and authorization from the spring app, and do this in decidated service in front of WDA: [Link to grant master plan]( https://github.com/flock-community/flock-eco-workday/blob/10f9305b524fbcfb2a7a7be96e6f4a0a0030268c/docs/identity-access-management.md)
image

### CODE REVIEW

File: .github/workflows/gcp.yml [hunk 1]
```diff
@@ -55,3 +55,12 @@ jobs:
...
       - name: Build and Deploy Development version
         run: ./mvnw -B compile -P develop -P frontend jib:build -Djib.container.environment=SPRING_PROFILES_ACTIVE=develop -Djib.to.image=gcr.io/flock-eco/flock-eco-workday-develop --file pom.xml
+
+      - id: build-and-push-oathkeeper
+        name: Build and push Oathkeeper
+        uses: docker/build-push-action@v3
+        with:
+          context: deployments/oathkeeper
+          push: true
+          tags: eu.gcr.io/flock-eco/flock-eco-workday-oathkeeper:develop
```
Comment (ZzAve) [inline .github/workflows/gcp.yml:65]: Do we wanna go 'eu' rather the us? Workday is send to `gcr.io`. I like the `eu.gcr.io` semantic, but not sure what we want here?

File: deployments/oathkeeper/rules/rules.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+- id: workday-everything
+  upstream:
+    url: https://host.docker.internal:8080
+  version: v0.40.3
+  match:
+    url: https://workday.flock.community/<.*>
```
Comment (ZzAve) [inline deployments/oathkeeper/rules/rules.yaml:6]: We need the upstream url of workday, which is the default url it gets assigned when the google cloud run was created. Something like: `https://workday-app-tveu425laa-ew.a.run.app`

File: deployments/oathkeeper/rules/rules.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - handler: noop
+  errors:
+    - handler: json
+
+- id: shadow-workday-everything
+  upstream:
+    url: https://host.docker.internal:8080
+  version: v0.40.3
+  match:
+    url: https://shadow.workday.flock.community/<.*>
```
Comment (ZzAve) [inline deployments/oathkeeper/rules/rules.yaml:26]: We need the upstream url of the shadow version of workday, which is the default url it gets assigned when the google cloud run was created. Something like: `https://shadow-workday-app-tveu425laa-ew.a.run.app`.

Comment (ZzAve) [reply deployments/oathkeeper/rules/rules.yaml]: Updated with `https://workday-app-ory-x6nsuevzyq-ew.a.run.app/` and `ory.workday.flock.community`

File: deployments/oathkeeper/rules/rules.yaml (hunk 2 above)
Comment (ZzAve) [reply deployments/oathkeeper/rules/rules.yaml]: Updated with `https://workday-app-x6nsuevzyq-ew.a.run.app/`

**Review by wilmveel — APPROVED** [review]

**Pull request merged**